### DISCUSSION TITLE
Is swap space configurable?

### DISCUSSION THREAD

**Discussion by qcloutier-sonrai** [lima-vm/lima#941] [Q&A]
I might be missing something, but I didn't see an option for it in the default.yaml file.

**Comment by afbjorklund** [comment]
Somehow a swap partition feels like a less priority than a data partition, but anyway.

I suppose one could use a swap **file**, but swapping in a virtual machine sounds strange...

ZRAM would be nice, though. (swapping to compression)

* https://github.com/lima-vm/alpine-lima/issues/41

**Reply by 13steinj** [reply]
Is there a workaround for this? could one ssh into the lima/colima VM, `modprobe zram` and set up swap or is that not possible in current state?

**Comment by afbjorklund** [comment]
Searching came up with:
https://www.redhat.com/sysadmin/cloud-swap

With the sample cloud-init:
```yaml
swap:
    filename: /swapfile
    size: auto
    maxsize: 4294967296

```

**Reply by qcloutier-sonrai** [reply]
Can this be configured for the Ubuntu image used by Rancher Desktop through the overrides file?

**Reply by afbjorklund** [reply]
I think it might need some feature implemented, before this will work. The template isn't "freeform" like that.

[pkg/cidata/cidata.TEMPLATE.d/user-data](https://github.com/lima-vm/lima/blob/master/pkg/cidata/cidata.TEMPLATE.d/user-data)

**Reply by jandubois** [reply]
Note that Rancher Desktop uses Alpine and not Ubuntu.

Setting swap via cloud-init would require support from Lima first, as noted above.

It might be possible to enable it using a provisioning script though.

For Alpine you would want to choose `/tmp/swapfile` instead of `/swapfile` because the root filesystem is RAM based, so adding a swapfile there would be counterproductive.

Also remember that kubernetes strongly recommends to disable swap on all nodes.

**Reply by afbjorklund** [reply]
They did actually relax the swap off requirement, with the Kubernetes 1.24 release (beta)

https://kubernetes.io/blog/2021/08/09/run-nodes-with-swap-alpha/

https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2400-node-swap

For previous k8s releases, one had to use `--ignore-preflight-errors Swap` to bypass

**Reply by qcloutier-sonrai** [reply]
In my case, I have Rancher's built-in cluster disabled. Maybe it would make sense to have the options be mutually exclusive? Not sure what Docker Desktop's behaviour is here.
